pkg/reconciler/configmap/configmap.go

```diff
@@ -0,0 +1,396 @@
+	configMap := corev1.ConfigMap{}
+	err := r.client.Get(ctx, request.NamespacedName, &configMap)
+	if err != nil {
+		return reconcile.Result{}, nil
+	}
+	enabled := configMap.Data[EnableRebuilds] == "true"
+	if enabled {
+		log.Info("Setup Rebuilds %s", "name", request.Name)
+		err := r.setupRebuilts(ctx, request)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	log.Info("Setup Cache %s", "name", request.Name)
+	if err = r.setupCache(ctx, request, enabled, configMap); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileConfigMap) setupCache(ctx context.Context, request reconcile.Request, rebuildsEnabled bool, configMap corev1.ConfigMap) error {
+	cache := v1.Deployment{}
+	deploymentName := types.NamespacedName{Namespace: request.Namespace, Name: CacheDeploymentName}
+	err := r.client.Get(ctx, deploymentName, &cache)
+	create := false
+	if err != nil {
+		if errors.IsNotFound(err) {
+			create = true
+			cache.Name = deploymentName.Name
+			cache.Namespace = deploymentName.Namespace
+			var replicas int32
+			replicas = 1
+			cache.Spec.Replicas = &replicas
+			cache.Spec.Selector = &v12.LabelSelector{MatchLabels: map[string]string{"app": CacheDeploymentName}}
+			cache.Spec.Template.ObjectMeta.Labels = map[string]string{"app": CacheDeploymentName}
+			cache.Spec.Template.Spec.Containers = []corev1.Container{{
+				Name:            CacheDeploymentName,
+				ImagePullPolicy: "Always",
+				SecurityContext: &corev1.SecurityContext{RunAsUser: i64a(0)},
+				Ports: []corev1.ContainerPort{{
+					Name:          "http",
+					ContainerPort: 8080,
+					Protocol:      "TCP",
+				}},
+				Resources: corev1.ResourceRequirements{
+					//TODO: make configurable
+					Requests: map[corev1.ResourceName]resource.Quantity{"memory": resource.MustParse("128Mi"), "cpu": resource.MustParse("250m")},
+					Limits:   map[corev1.ResourceName]resource.Quantity{"memory": resource.MustParse("700Mi"), "cpu": resource.MustParse("500m")},
+				},
+			}}
+			cache.Spec.Template.Spec.ServiceAccountName = ServiceAccountName
+
+		} else {
+			return err
+		}
+	}
+
+	//and setup the service
+	err = r.client.Get(ctx, types.NamespacedName{Name: CacheDeploymentName, Namespace: configMap.Namespace}, &corev1.Service{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			service := corev1.Service{
+				ObjectMeta: ctrl.ObjectMeta{
+					Name:      CacheDeploymentName,
+					Namespace: configMap.Namespace,
+				},
+				Spec: corev1.ServiceSpec{
+					Ports: []corev1.ServicePort{
+						{
+							Name:       "http",
+							Port:       80,
+							TargetPort: intstr.IntOrString{IntVal: 8080},
+						},
+					},
+					Type:     corev1.ServiceTypeClusterIP,
+					Selector: map[string]string{"app": CacheDeploymentName},
+				},
+			}
+			err := r.client.Create(ctx, &service)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	type Repo struct {
+		name     string
+		position int
+	}
+	//TODO: make configurable
```

Comment:
I suppose my question from https://github.com/redhat-appstudio/jvm-build-service/pull/105/files#r932615502 applies here too ... need a mem refresh from one of the voice to voice chats we had last week.